tools/codegen/gen.py

```diff
@@ -276,10 +276,21 @@ def func(f: NativeFunction) -> Optional[str]:
                     cuda_guard = """\
     // DeviceGuard omitted
 """
+            mem_overlap_check = ""
+            if (dispatch == "CPU" or dispatch == "CUDA") and ('backward' not in impl_name):
+                if f.func.name.name.inplace:
+                    mem_overlap_check = """\
+    at::assert_no_internal_overlap(self);
```

Comment:
What if the function being called can handle internal memory overlap? Then we would want a mechanism to disable this for those functions.

One question I have is, in the long-term, do we want our implementations of operators to handle internal memory overlap or should we just disable these operations.

Reply:
I could add a list of operations that should not perform this check. This would also apply for in-place operations that do not modify storage data, like `requires_grad_()`

Reply:
Another option might be to add some kind of flag to `native_functions.yaml` to turn off the check